superdesk/publish_async/exchanges/content_exchange.py

```diff
@@ -0,0 +1,232 @@
+                processing the request. Sets the queue state to pending and adds relevant error messages.
+        :raises Exception: For all other unexpected errors during processing. Sets the queue state
+                to error and logs appropriate messages to help debugging.
+        """
+
+        # Update the ``published`` collection, to update it's state
+        published_service = get_resource_service(PUBLISHED)
+
+        published_item = published_service.find_one(
+            req=None, item_id=request.item_id, _current_version=request.item[VERSION]
+        )
+        if not published_item:
+            # If we failed to get the item by ``_current_version``, then try ``last_published_version`` instead
+            logger.warning(f"Published item {request.item_id} not found.")
+
+            published_item = published_service.find_one(req=None, item_id=request.item_id, last_published_version=True)
+            if not published_item:
+                logger.warning(
+                    f"Published item {request.item_id} not found in either ``last_published_version`` or ``_current_version``."
+                )
+            return PublishRequestResponse(routed=False)
+
+        published_item_id = published_item[ID_FIELD]
+
+        logger.info(f"Published item QUEUE_STATE == {published_item.get(QUEUE_STATE)}")
+
+        if self.polling and published_item.get(QUEUE_STATE) == PublishState.PUSHED:
+            # This request will be processed by ``PublishExchangeFactory.send_scheduled_or_pending_content``
+            logger.info(f"Setting item {request.item_id} to be polled for later")
+            await self.set_published_item_pending(published_item_id)
+            return PublishRequestResponse(routed=True)
+
+        try:
+            if request.item.get(ITEM_STATE) == CONTENT_STATE.SCHEDULED:
+                if (utc_schedule := get_utc_publish_schedule(published_item)) and utc_schedule < utcnow():
+                    # This item will be picked up by the ``ExchangeFactory.send_scheduled_or_pending_content`` task
+                    # So we respond that the item was routed, i.e. success (for now)
+                    return PublishRequestResponse(routed=True)
+
+                await self.updated_scheduled_item(published_item)
+            else:
+                await self.update_published_item(published_item_id)
+
+            response = await super().send(request)
+
+            if not response.content_api_subscribers and request.publish_to_content_api and content_api.is_enabled():
+                try:
+                    # If there were no ContentAPI Subscribers, we push it there manually now
+                    get_resource_service("content_api").publish(request.item, [])
+                except Exception:
+                    logger.exception(
+                        f"Failed to queue item to API for item: {request.item_id} for action {request.operation}"
```

Comment:
would be good to avoid unique log messages so sentry can group those, you can put item_id and operation in `extra` param